### PR TITLE
[ONNX] GatherND Support for Opset 11

### PR DESCRIPTION
**Pull request by jwfromm**
This PR adds a converter for the ATen Index operator in Onnx opsets 11+ that uses GatherND rather than the hackier approach used in Opset 9. Although I've confirmed this implementation works well using the onnx runtime on output graphs, it's a little unclear to me how Onnx testing is meant to work within Pytorch. Not only are the onnx tests in `test_operators.py` failing on the master branch (before the changes in this PR), but it seems like the `expect` files used for testing are generated by the tests themselves which seems to defeat the purpose. I'd appreciate some pointers on how to make this testing more robust.

### CODE REVIEW

**Comment by jwfromm** [comment]
@BowenBao can you take a look at this PR?

**Review by houseroad — COMMENTED** [review]
Tests are failing. Could you fix them?

**Comment by jwfromm** [comment]
@houseroad, I'm a little blocked on the test for the new `gather_nd` operator because it's unclear to me how `expect` files are meant to be generated. Can you point me to where they're coming from?

**Comment by jwfromm** [comment]
@houseroad, I believe I've fixed the tests for the new `gather_nd` conversion but can't figure out why CI is complaining about `test_dropout_training_opset12`. It passes locally and does not contain any `aten::index` calls. Do you have any ideas how to proceed?

File: test/onnx/test_operators.py [hunk 1]
```diff
@@ -681,12 +685,6 @@ def test_dropout_training(self):
         x = torch.randn(3, 4, requires_grad=True)
         self.assertONNX(lambda x: torch.max(functional.dropout(x)), x, training=torch.onnx.TrainingMode.TRAINING)
 
-    @unittest.skip("disable test until onnx submodule is updated")
```
Comment (BowenBao) [inline test/onnx/test_operators.py:684]: This is probably the reason why CI failing. I suppose they are removed by accident?

**Review by BowenBao — COMMENTED** [review]
@jwfromm Thanks for upgrading the symbolic for index export! Please see my comments inline.

The CI failing on dropout because it is depending on newer version of ONNX. This test will be enabled after PyTorch ONNX submodule update. 

The more general place of adding PyTorch - ONNX tests are these two files
https://github.com/pytorch/pytorch/blob/master/test/onnx/test_pytorch_onnx_caffe2.py
https://github.com/pytorch/pytorch/blob/master/test/onnx/test_pytorch_onnx_onnxruntime.py

For this PR you should add your tests in `test_pytorch_onnx_onnxruntime.py`, since caffe2 does not currently support opset 11.

Also note since the CI is failing early on dropout, a lot of tests regarding index in `test_pytorch_onnx_onnxruntime.py` is not being run.

**Comment by jwfromm** [comment]
Thanks for all the pointers @BowenBao, looking at and running `test_pytorch_onnx_onnxruntime.py` is definitely what I should have been doing from the start. Unfortunately, there are some good test cases there that this implementation can't handle. I'm going to close this PR as I don't have a solution for some of the edge cases. I really appreciate your pointers and feedback though!
